packages/gatsby/src/utils/sanitize-node.js

```diff
@@ -14,6 +14,16 @@ const sanitizeNode = (data, isNode = true, path = new Set()) => {
     path.add(data)
 
     const returnData = isPlainObject ? {} : []
+    // Refer to https://github.com/gatsbyjs/gatsby/issues/26565
```

Comment:
[_.each](https://lodash.com/docs/4.17.15#forEach) is work for both `arrary` and `object` and show high performance, so that's why gatsbyjs use it at here. For preventing  `OOM` when we use `_.each` to loop `{ length:  SUPER_MAX_INTEGER, ... }` object, we also can refact these code to `Object.keys` or other implemention. It's depends on the opinion of the gatsby team.